### ISSUE TITLE
Terragrunt should detect if backend configuration is missing from Terraform code

### ISSUE THREAD

**Issue by brikis98** [gruntwork-io/terragrunt#212]
As of Terraform 0.9, you have to declare a `backend` in your Terraform code, or remote state won't work:

```hcl
terraform {
  # The configuration for this backend will be filled in by Terragrunt
  backend "s3" {}
}
```

This is very easy to miss. See #211, #207, and #183. 

If the user specifies a `remote_state` configuration in their Terragrunt configuration, Terragrunt should scan the user's `.tf` files and report an error if a `backend` block is not found. This could be done using an approach similar to how we look for modules: https://github.com/gruntwork-io/terragrunt/blob/master/cli/cli_app.go#L254

**Comment by hSATAC** [comment]
Thanks @brikis98 man you just saved my day. 💯

**Comment by mazubieta** [comment]
Please include this, soon.  I just unwittingly deleted my local state files thinking that I had properly setup remote state.

Thankfully, I was able to recover via backups.

**Comment by brikis98** [comment]
This should be fixed by #302. New binaries will be in https://github.com/gruntwork-io/terragrunt/releases/tag/v0.13.5 shortly.
